cinder/volume/drivers/vmware/volumeops.py

```diff
@@ -995,7 +995,15 @@ def get_datastore_properties(self, datastore):
                                         properties_to_collect=[
                                             "summary",
                                             "availableField",
-                                            "customValue"
+                                            "customValue",
+                                            "triggeredAlarmState"
+                                        ])
+
+    def get_datastore_alarm(self, alarm):
+        return self._session.invoke_api(vim_util, 'get_object_properties_dict',
+                                        self._session.vim, alarm,
+                                        properties_to_collect=[
+                                            'info'
```

Comment:
I don't get that part. We retrieve `info` but up when using it, we access `alarm_info['description']`. So does that fetch everything under `info` into a dict or is it a typo or how does it work?

Reply:
Fixed

Reply:
I see now, that we actually need `description` of the `info` - could we then directly request `info.description` and not fetch all the `info`? This should speed up the calls as we need to parse and transfer less XML. We then have to update the accesses above to also access `info.description` instead of `['info']['description']`.